TODO.md

```diff
@@ -2,14 +2,8 @@
 * organize this file (at least sort... somehow)
 * consider adding a struct when faced with long argument lists (for methods) (e.g., recursive_join)
 * return `None` vs `Err`
-* fail early on malformed queries (e.g., anything outside special chars and A-Z, a-z (esp. outside ASCII 0-127))
-* avoid duplicating work (e.g., `parse_form` call in `make_list`)
-* add detailed error messages
```

Comment:
technically correct that we (well, you) have added detailed error messages, but pretty sure there are more to go so consider leaving this one here